src/base/format_impl.h

```diff
@@ -0,0 +1,920 @@
+		arg_.unsigned_val = t;
+		format_do_impl_run(out, orig_index, localize, AbstractNode::ArgType::kUnsigned);
+	}
+	inline void
+	format_do_impl(char** out, unsigned orig_index, bool localize, unsigned long int t) const {
+		arg_.unsigned_val = t;
+		format_do_impl_run(out, orig_index, localize, AbstractNode::ArgType::kUnsigned);
+	}
+	inline void
+	format_do_impl(char** out, unsigned orig_index, bool localize, unsigned int t) const {
+		arg_.unsigned_val = t;
+		format_do_impl_run(out, orig_index, localize, AbstractNode::ArgType::kUnsigned);
+	}
+	inline void
+	format_do_impl(char** out, unsigned orig_index, bool localize, unsigned short t) const {
+		arg_.unsigned_val = t;
+		format_do_impl_run(out, orig_index, localize, AbstractNode::ArgType::kUnsigned);
+	}
+	inline void format_do_impl(char** out, unsigned orig_index, bool localize, uint8_t t) const {
+		arg_.unsigned_val = t;
+		format_do_impl_run(out, orig_index, localize, AbstractNode::ArgType::kUnsigned);
+	}
+
+	inline void format_do_impl(char** out, unsigned orig_index, bool localize, const void* t) const {
+		arg_.unsigned_val = reinterpret_cast<uintptr_t>(t);
+		format_do_impl_run(out, orig_index, localize, AbstractNode::ArgType::kPointer);
+	}
+	inline void
+	format_do_impl(char** out, unsigned orig_index, bool localize, const std::nullptr_t) const {
+		format_do_impl_run(out, orig_index, localize, AbstractNode::ArgType::kNullptr);
+	}
+
+	template <typename T>
+	inline bool format_impl(
+	   char** out, unsigned orig_index, unsigned arg_index, const bool localize, T t) const {
+		if (arg_index != 0) {
+			throw wexception("Too few arguments provided: %u required but only %u passed",
+			                 format_nodes_count_, orig_index - arg_index);
+		}
+		format_do_impl(out, orig_index, localize, t);
+		return true;
+	}
+	template <typename T, typename... Args>
+	inline bool format_impl(char** out,
+	                        unsigned orig_index,
+	                        unsigned arg_index,
+	                        const bool localize,
+	                        T t,
+	                        Args... args) const {
+		if (arg_index == 0) {
+			format_do_impl(out, orig_index, localize, t);
+			return false;
+		}
+		// By the way, this is not a recursive call – the compiler generates a separate
+		// function for every individual template specification, and they're all inlined.
+		return format_impl(out, orig_index, arg_index - 1, localize, args...);
+	}
+
+	// Parser functions
+	explicit Tree(const char* format_string);
+	void parse_next_node(const char*&);
+	void parse_literal_node(const char*&);
+	void parse_format_index(const char*&);
+	bool parse_index_or_minwidth(const char*&);
+	void parse_flags_and_minwidth(const char*&);
+	void parse_precision(const char*&);
+	std::unique_ptr<AbstractNode> parse_type_spec(const char*&);
+	void recognized_wildcard_node();
+	void recognized_node(std::unique_ptr<AbstractNode>);
```

Comment:
So these have to be here as member functions to be able to access member variables. But they are implemented as inline for efficiency. So they can't be called by anything outside `Tree()`, or bad things would happen. Ouch, and sorry, and someone more informed than me should sort this out, but I still like this new modular `Tree()` with its own .cc file, and `format_impl.h` is still way too big...